### PR TITLE
Cleanup quadrature

### PR DESCRIPTION
**Pull request by drwells**
I started by fixing another `-Wtype-limits` warning and then did some other cleaning up.

### CODE REVIEW

File: source/base/quadrature.cc [hunk 1]
```diff
@@ -107,6 +108,17 @@ Quadrature<1>::Quadrature(const Point<1> &point)
 
 
 
+template <>
+Quadrature<0>::Quadrature(const Point<0> &point)
+  : quadrature_points(std::vector<Point<0>>(1, point))
+  , weights(std::vector<double>(1, 1.))
```
Comment (tjhei) [inline source/base/quadrature.cc:114]: I think you can skip all these initializations.

File: source/base/quadrature.cc [hunk 2]
```diff
@@ -449,108 +456,6 @@ namespace internal
   }   // namespace QIteratedImplementation
 } // namespace internal
 
-// template <>
```
Comment (tjhei) [inline source/base/quadrature.cc:452]: huh

Comment (drwells) [reply source/base/quadrature.cc:452]: This function does almost the same thing as the constructor which is why we got rid of it. Are you asking for a change?

File: source/base/quadrature.cc (hunk 1 above)
Comment (drwells) [reply source/base/quadrature.cc:114]: Good point. We do need to initialize the boolean though.

**Review by bangerth — APPROVED** [review]
I think the only functional change is the addition of the `Quadrature(const Point<0> &)` constructor, right? Everything else is just removing commented-out lines and reflowing comments. I wonder then why you actually need the new constructor?

**Comment by drwells** [comment]
I need the constructor to avoid the loop in the dim = 0 case to avoid a compiler warning.

**Pull request merged**